dask_deltatable/core.py

```diff
@@ -1,34 +1,34 @@
 import json
 import os
-from typing import Dict, List, Optional
+from typing import Any, Dict, List, Optional
 from urllib.parse import urlparse
 
 import dask
 import dask.dataframe as dd
-import pyarrow.parquet as pq
+import pyarrow.parquet as pq  # type: ignore[import]
 from dask.base import tokenize
 from dask.dataframe.io import from_delayed
 from dask.delayed import delayed
 from deltalake import DataCatalog, DeltaTable
-from fsspec.core import get_fs_token_paths
+from fsspec.core import get_fs_token_paths  # type: ignore[import]
 from pyarrow import dataset as pa_ds
 
 
 class DeltaTableWrapper:
     path: str
-    version: int
-    columns: List[str]
-    datetime: str
-    storage_options: Dict[str, any]
+    version: Optional[int]
```

Comment:
I think it would be better to use ``int | None`` (this is the new syntax)

Reply:
Also dict instead of Dict, list instead of List, ...

Reply:
This is still supporting python 3.8 without `from __future__ import annotations`

Reply:
Any reason not to import from future? This looks good to me if there are reasons against this

Reply:
I added `from __future__ import annotations` and pyupgrade fixed the calls to `List`/`Dict`/`Optional` but I needed to add flake8 to detect the unused imports